plugins/autocomplete/plugin.js

```diff
@@ -0,0 +1,1422 @@
+				if ( itemId == null ) {
+					return;
+				}
+			}
+
+			var item = this.model.getItemById( itemId ),
+				editor = this.editor;
+
+			editor.fire( 'saveSnapshot' );
+			editor.getSelection().selectRanges( [ this.model.range ] );
+			editor.insertHtml( this.getHtmlToInsert( item ), 'text' );
+			editor.fire( 'saveSnapshot' );
+		},
+
+		/**
+		 * Destroys the autocomplete instance.
+		 * View element and event listeners will be removed from the DOM.
+		 */
+		destroy: function() {
+			CKEDITOR.tools.array.forEach( this._listeners, function( obj ) {
+				obj.removeListener();
+			} );
+
+			this._listeners = [];
+
+			this.view.element.remove();
+		},
+
+		/**
+		 * Returns HTML that should be inserted into the editor when the item is committed.
+		 *
+		 * See also the {@link #commit} method.
+		 *
+		 * @param {CKEDITOR.plugins.autocomplete.model.item} item
+		 * @returns {String} The HTML to insert.
+		 */
+		getHtmlToInsert: function( item ) {
+			var encodedItem = encodeItem( item );
+			return this.outputTemplate ? this.outputTemplate.output( encodedItem ) : encodedItem.name;
+		},
+
+		/**
+		 * Creates and returns the model instance. This method is used while
+		 * initializing the autocomplete and can be overwritten in order to
+		 * return an instance of a different class than the default model.
+		 *
+		 * @param {Function} dataCallback See {@link CKEDITOR.plugins.autocomplete} arguments.
+		 * @returns {CKEDITOR.plugins.autocomplete.model} The model instance.
+		 */
+		getModel: function( dataCallback ) {
+			return new Model( dataCallback );
+		},
+
+		/**
+		 * Creates and returns the text watcher instance. This method is used while
+		 * initializing the autocomplete and can be overwritten in order to
+		 * return an instance of a different class than the default text watcher.
+		 *
+		 * @param {Function} textTestCallback See {@link CKEDITOR.plugins.autocomplete} arguments.
+		 * @returns {CKEDITOR.plugins.textWatcher} The text watcher instance.
+		 */
+		getTextWatcher: function( textTestCallback ) {
+			return new CKEDITOR.plugins.textWatcher( this.editor, textTestCallback, this.throttle );
+		},
+
+		/**
+		 * Creates and returns the view instance. This method is used while
+		 * initializing the autocomplete and can be overwritten in order to
+		 * return an instance of a different class than the default view.
+		 *
+		 * @returns {CKEDITOR.plugins.autocomplete.view} The view instance.
+		 */
+		getView: function() {
+			return new View( this.editor );
+		},
+
+		/**
+		 * Opens the panel if {@link CKEDITOR.plugins.autocomplete.model#hasData there is any data available}.
+		 */
+		open: function() {
+			if ( this.model.hasData() ) {
+				this.model.setActive( true );
+				this.view.open();
+				this.model.selectFirst();
+				this.view.updatePosition( this.model.range );
+			}
+		},
+
+		// LISTENERS ------------------
+
+		/**
+		 * The function registered by the {@link #attach} method as the
```

Comment:
This function should be marked as private.

Please do that for other listeners.

Reply:
It also should document what the method does instead of who is using it.

E.g. Function to be called when once content (?) has changed.

I started to wonder, shouldn't we rename it to `reposition`? 🤔 I think it would be a better name. Then again, it's used not only when content changes, but also for scrolls.